package ch08;

public class Test {

	Exception e;
	
	public static void main(String[] args) {
		// TODO Auto-generated method stub

		/*
		 * Arithmetic Exception :정수를 0 으로 나눈 경우 발생
			NullPointerException : 초기화 되지 않은 Object를 사용하는 경우
			Dog d = null;
			System.out.println(dog);
			
			ArrayIndexOutOfBoundsException :배열의 크기를 넘어선 위치를 참조하려는 경우
			FileNotFoundException :참조하는 파일이 지정된 위치에 존재하지 않는 경우
			ClassNotFoundException :
			Class.forName(“sis.studentinfo.Student”); //클래스가 로드되지 않은 경우
			InterruptedException :Thread.sleep(), join(). Object의 wait()로 non-runnable 상태인 thread를 Runnable하게 만들 수 있도록 사용할 수 있음
		 */
	}

}
